core/src/main/java/org/testcontainers/containers/wait/HostPortWaitStrategy.java

```diff
@@ -33,7 +33,7 @@ protected void waitUntilReady() {
         Callable<Boolean> checkStrategy;
 
         // Special case for Docker for Mac, see #160
-        if (DockerClientFactory.instance().isUsing(ProxiedUnixSocketClientProviderStrategy.class)) {
+        if (System.getProperty("os.name").toLowerCase().contains("mac")) {
```

Comment:
only Docker for Mac's port checking is broken, this is why we check here. Docker Machine works fine